autopush/db.py

```diff
@@ -59,17 +60,31 @@ def create_rotating_message_table(prefix="message", read_throughput=5,
 
 
 def get_rotating_message_table(prefix="message", delta=0):
-    """Gets the message table for the current month.
-
-    This requires the table to already exist or errors will occur.
-
-    """
-    return Table(make_rotating_tablename(prefix, delta))
+    """Gets the message table for the current month."""
+    db = DynamoDBConnection()
+    dblist = db.list_tables()["TableNames"]
+    tablename = make_rotating_tablename(prefix, delta)
+    if tablename not in dblist:
+        return create_rotating_message_table(prefix=prefix, delta=delta)
+    else:
+        return Table(tablename)
 
 
 def create_router_table(tablename="router", read_throughput=5,
                         write_throughput=5):
-    """Create a new router table"""
+    """Create a new router table
+
+    The last_connect index is a value used to determine the last month a user
+    was seen in. To prevent hot-keys on this table during month switchovers the
+    key is determined based on the following scheme:
+
+        (YEAR)(MONTH)(DAY)(HOUR)(0001-0010)
+
+    Note that the random key is only between 1-10 at the moment, if the key is
+    still too hot during production the random range can be increased at the
+    cost of additional queries during GC to locate expired users.
```

Comment:
I'm a bit confused by this. last_connect is set to the YYMMDDHHrr value, but it's not a hash or range key, so it's an attribute. In has_connected_this_month, we only check for YYMM. Even in the message table, there's another attribute that records just the Month along with the last_connect. 

Why the high definition last_connect if we're throwing away more than half of it? Wouldn't YYMMr+